java-frontend/src/test/files/se/SimpleAssignmentExecution.java

```diff
@@ -0,0 +1,10 @@
+class A {
+
+  int f;
+
+  void compoundAssignment(int a) {
+    a = 1;
+    this.f = 1;
+    some.complex().lhs = 1;
```

Comment:
this one 3rd case is failing with my modification, and sounds very strange to me. I'm not confident it is related to our issue, and to me it is related as how we handle field of objects which are not "this" or "super".